arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "readOnly": true
+        }
+      }
+    },
+    "AsyncOperationResult": {
+      "description": "Contains properties related to the status of a long running operation.",
+      "type": "object",
+      "properties": {
+        "Status": {
+          "description": "Gets or sets the status.",
+          "type": "string"
+        },
+        "Error": {
+          "$ref": "#/definitions/ErrorMesssage",
+          "description": "Gets or sets the error."
+        }
+      }
+    },
+    "ErrorMesssage": {
+      "description": "Contains properties relates to an error message.",
+      "type": "object",
+      "properties": {
+        "Code": {
+          "description": "Gets or sets the code.",
+          "type": "string"
+        },
+        "Message": {
+          "description": "Gets or sets the message.",
+          "type": "string"
+        }
+      }
+    },
+    "RegistryStatistics": {
+      "description": "Contains the properties related to the registry statistics of an IotHub.",
+      "type": "object",
+      "properties": {
+        "totalDeviceCount": {
+          "format": "int64",
+          "description": "Gets or sets the total device count.",
+          "type": "integer"
+        },
+        "enabledDeviceCount": {
+          "format": "int64",
+          "description": "Gets or sets the enabled device count.",
+          "type": "integer"
+        },
+        "disabledDeviceCount": {
+          "format": "int64",
+          "description": "Gets or sets the disabled device count.",
+          "type": "integer"
+        }
+      }
+    },
+    "JobResponseListResult": {
+      "description": "Contains the user view of a List Result with a next link",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Gets or sets the list of",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobResponse"
+          }
+        },
+        "nextLink": {
+          "description": "Gets the next link.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "IotHubSkuDescription": {
+      "description": "Contains properties related to the Iot Hub Sku.",
+      "type": "object",
+      "properties": {
+        "resourceType": {
+          "description": "Gets the type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "sku": {
+          "$ref": "#/definitions/IotHubSkuInfo",
+          "description": "Gets or sets the sku."
+        },
+        "capacity": {
+          "$ref": "#/definitions/IotHubCapacity",
+          "description": "Gets or sets the capacity."
+        }
+      }
+    },
+    "IotHubCapacity": {
+      "description": "Contains the properties related to the IotHub capacity information.",
+      "type": "object",
+      "properties": {
+        "minimum": {
+          "format": "int64",
+          "description": "Gets or sets the minimum number of units.",
```

Comment:
Are there any constraints on the min max value that the user can provide? If so then you should set them as validation constraints and the generated code will validate against that.

Take a look over [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#patterned-fields-2). Once the page is loaded scroll a little up and you will see the list of supported constraints in swagger.